client/src/Tickets/TicketModal.jsx

```diff
@@ -84,26 +86,78 @@ function Charger({ stateChange }) {
   );
 }
 
-const AddTicketModel = ({ update, data, stateChange, toggleUserModal, setToggleUserModal }) => {
+const TicketModal = ({ onCreate, onUpdate }) => {
+  const navigate = useNavigate();
+  const { ticketId } = useParams();
+  const [data, setData] = useState({
+    ClientId: null,
+    LocationId: null,
+    UserId: null,
+    serialNumber: '',
+    device: '',
+    problem: '',
+    troubleshooting: '',
+    resolution: '',
+    dateOn: DateTime.now().toISODate(),
+    timeInAt: '',
+    timeOutAt: '',
+    hasCharger: false,
+    notes: '',
+  });
+
+  useEffect(() => {
+    async function fetchData() {
+      const response = await fetch(`/api/tickets/${ticketId}`);
+      if (response.ok) {
+        const data = await response.json();
+        setData(data);
+      }
+    }
+    if (ticketId) {
+      fetchData();
+    }
+  }, [ticketId]);
+
+  const onChange = (e) => {
+    const newData = { ...data };
+    newData[e.target.name] = e.target.value;
+    setData(newData);
+  };
+
   const submitTicket = async (e) => {
     e.preventDefault();
-    const result = await fetch('/api/tickets', {
-      method: 'POST',
-      headers: {
-        'Content-Type': 'application/json',
-      },
-      body: JSON.stringify(data),
-    });
-    result.json().then((ticket) => {
-      ticket['createdAt'] = DateTime.fromISO(ticket['createdAt']).toLocaleString();
-      update(ticket);
-    });
-
-    setToggleUserModal(false);
+    let response;
+    if (data.id) {
+      response = await fetch(`/api/tickets/${data.id}`, {
+        method: 'PATCH',
+        headers: {
+          'Content-Type': 'application/json',
+        },
+        body: JSON.stringify(data),
+      });
+    } else {
+      response = await fetch('/api/tickets', {
+        method: 'POST',
+        headers: {
+          'Content-Type': 'application/json',
+        },
+        body: JSON.stringify(data),
+      });
+    }
+    if (response.ok) {
```

Comment:
Note that the response code should always be checked so that errors can be handled...